clang/lib/Driver/OffloadBundler.cpp

```diff
@@ -178,8 +178,7 @@ bool OffloadTargetInfo::isOffloadKindCompatible(
 }
 
 bool OffloadTargetInfo::isTripleValid() const {
-  return !Triple.str().empty() && (Triple.getArch() != Triple::UnknownArch ||
-                                   Triple.str() == "native_cpu---");
+  return !Triple.str().empty() && Triple.getArch() != Triple::UnknownArch;
```

Comment:
Do we have to check both?

Thanks

Reply:
I think you're right, we don't need to. This restores us to how it is written in LLVM upstream though (https://github.com/llvm/llvm-project/blob/main/clang/lib/Driver/OffloadBundler.cpp#L137-L139, originally added by https://reviews.llvm.org/D93525 and not caught in review there) so minimizes the diff we have with upstream.